src/main/java/bio/terra/cli/service/utils/WorkspaceManagerService.java

```diff
@@ -332,4 +365,297 @@ public RoleBindingList getRoles(UUID workspaceId) {
+                      .numNewerVersions(2));
+      lifecycleRules = Collections.singletonList(lifecycleRule);
+    }
+
+    ControlledGcpResourceApi controlledGcpResourceApi = new ControlledGcpResourceApi(apiClient);
+    CreateControlledGcpGcsBucketRequestBody createRequest =
+        new CreateControlledGcpGcsBucketRequestBody()
+            .common(
+                new ControlledResourceCommonFields()
+                    .name(name)
+                    .description(description)
+                    .cloningInstructions(cloningInstructions)
+                    .accessScope(accessScope)
+                    .managedBy(ManagedBy.USER))
+            .gcsBucket(
+                new GcpGcsBucketCreationParameters()
+                    .name(gcsBucketName)
+                    .defaultStorageClass(defaultStorageClass)
+                    .lifecycle(new GcpGcsBucketLifecycle().rules(lifecycleRules))
+                    .location(location));
+    try {
+      return controlledGcpResourceApi.createBucket(createRequest, workspaceId).getGcpBucket();
+    } catch (ApiException ex) {
+      throw new SystemException("Error creating controlled GCS bucket in the workspace.", ex);
+    }
+  }
+
+  /**
+   * Call the Workspace Manager POST
+   * "/api/workspaces/v1/{workspaceId}/resources/controlled/gcp/bqdatasets" endpoint to add a Big
+   * Query dataset as a controlled resource in the workspace.
+   *
+   * @param workspaceId the workspace to add the resource to
+   * @param name name of the resource. this is unique across all resources in the workspace
+   * @param description description of the resource
+   * @param cloningInstructions instructions for how to handle the resource when cloning the
+   *     workspace
+   * @param accessScope access to allow other workspaces users
+   * @param bigQueryDatasetId Big Query dataset id
+   *     (https://cloud.google.com/bigquery/docs/datasets#dataset-naming)
+   * @param location Big Query dataset location (https://cloud.google.com/bigquery/docs/locations)
+   * @return the Big Query dataset resource object
+   */
+  public GcpBigQueryDatasetResource createControlledBigQueryDataset(
+      UUID workspaceId,
+      String name,
+      String description,
+      CloningInstructionsEnum cloningInstructions,
+      AccessScope accessScope,
+      String bigQueryDatasetId,
+      @Nullable String location) {
+    ControlledGcpResourceApi controlledGcpResourceApi = new ControlledGcpResourceApi(apiClient);
+    CreateControlledGcpBigQueryDatasetRequestBody createRequest =
+        new CreateControlledGcpBigQueryDatasetRequestBody()
+            .common(
+                new ControlledResourceCommonFields()
+                    .name(name)
+                    .description(description)
+                    .cloningInstructions(cloningInstructions)
+                    .accessScope(accessScope)
+                    .managedBy(ManagedBy.USER))
```

Comment:
Hmm, not this PR, but eventually I think we also need to specify the `PrivateResourceUser` fields for private resources I guess all created resources right now are workspace shared? I think it's a bad request exception when `AccessScope` is `PRIVATE_ACCESS` but no `PrivateResourceUser` roles (and user) is defined.

Reply:
Good catch, I overlooked those fields. Added.